File: merchant-frontend/src/types/product.ts
export interface Product {
  id: number
  categoryId: number
  brandId: number
  name: string
  subtitle: string
  mainImage: string
  images: string[]
  description: string
  status: number
  skus: ProductSKU[]
  specs: ProductSpec[]
  createdAt: string
  updatedAt: string
}

export interface ProductSKU {
  id: number
  skuCode: string
  price: number
  originalPrice: number
  costPrice: number
  barCode: string
  specValues: string
  status: number
}

export interface ProductSpec {
  name: string
  values: string[]
}

export interface CreateProductReq {
  categoryId: number
  brandId: number
  name: string
  subtitle: string
  mainImage: string
  images: string[]
  description: string
  status: number
  skus: CreateSKUReq[]
  specs: ProductSpec[]
}

export interface CreateSKUReq {
  skuCode: string
  price: number
  originalPrice: number
  costPrice: number
  barCode: string
  specValues: string
  status: number
}

export type UpdateProductReq = CreateProductReq

export interface Category {
  id: number
  parentId: number
  name: string
  level: number
  sort: number
  icon: string
  status: number
  children?: Category[]
}

export interface CreateCategoryReq {
  parentId: number
  name: string
  level: number
  sort: number
  icon: string
  status: number
}

export interface Brand {
  id: number
  name: string
  logo: string
  status: number
}

export interface CreateBrandReq {
  name: string
  logo: string
  status: number
}

File: product/domain/product.go
package domain

import "time"

type Product struct {
	ID          int64
	TenantID    int64
	CategoryID  int64
	BrandID     int64
	Name        string
	Subtitle    string
	MainImage   string
	Images      string // JSON array
	Description string
	Status      ProductStatus
	Sales       int64
	SKUs        []SKU
	Specs       []ProductSpec
	Ctime       time.Time
	Utime       time.Time
}

type ProductStatus uint8

const (
	ProductStatusDraft       ProductStatus = 1
	ProductStatusPublished   ProductStatus = 2
	ProductStatusUnpublished ProductStatus = 3
)

type SKU struct {
	ID            int64
	TenantID      int64
	ProductID     int64
	SpecValues    string // JSON: {"颜色":"红","尺码":"XL"}
	Price         int64  // 分
	OriginalPrice int64
	CostPrice     int64
	SKUCode       string
	BarCode       string
	Status        SKUStatus
	Ctime         time.Time
	Utime         time.Time
}

type SKUStatus uint8

const (
	SKUStatusActive   SKUStatus = 1
	SKUStatusInactive SKUStatus = 2
)

type ProductSpec struct {
	ID        int64
	ProductID int64
	TenantID  int64
	Name      string // e.g. "颜色"
	Values    string // JSON array: ["红色","蓝色"]
}
